packages/server/src/updater.ts

```diff
@@ -0,0 +1,50 @@
+import * as https from "https";
+
+export const compareVersions = (a: string, b: string): number | undefined => {
+	if (a === b) {
+		return 0;
+	}
+	const regEx = /[^0-9.]/g;
+	const aSplit = a.replace(regEx, "").split(".");
+	const bSplit = b.replace(regEx, "").split(".");
+	const dist = Math.max(aSplit.length, bSplit.length);
+	for (let i = 0; i < dist; i++) {
+		const aVal = parseInt(aSplit[i], 10);
+		const bVal = parseInt(bSplit[i], 10);
+		if (aVal > bVal || isNaN(bVal)) {
+			return 1;
+		}
+		if (aVal < bVal || isNaN(aVal)) {
+			return -1;
+		}
+	}
+};
+
+export const getRecentRelease = (): Promise<any> => {
+	return new Promise<string>((resolve, rej) => {
+		const options = {
+			host: "api.github.com",
+			path: "/repos/codercom/code-server/releases/latest",
+			method: "GET",
+			agent: false,
+			headers: {
+				"User-Agent": "code-server",
+			},
+		};
+
+		https.get(options, function (res) {
+			if (res.statusCode !== 200) {
+				throw new Error("Failed to acquire release information");
```

Comment:
You're inside a Promise. Maybe you should let the caller knows by rejecting it?

```suggestion
				rej(new Error("Failed to acquire release information"));
```